Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Dashboard/Ride.hs

```diff
@@ -0,0 +1,48 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# LANGUAGE TypeApplications #-}
+
+module Domain.Action.Dashboard.Ride
+  ( shareRideInfo,
+  )
+where
+
+import qualified "dashboard-helper-api" Dashboard.Common as Common
+import qualified "dashboard-helper-api" Dashboard.RiderPlatform.Ride as Common
+import qualified Domain.Action.UI.Ride as DRide
+import qualified Domain.Types.Merchant as DM
+import qualified Domain.Types.Ride as DRide
+import Environment
+import Kernel.Prelude
+import Kernel.Storage.Esqueleto.Transactionable (runInReplica)
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import SharedLogic.Merchant (findMerchantByShortId)
+import qualified Storage.Queries.Booking as QBooking
+import qualified Storage.Queries.Ride as QRide
+
+---------------------------------------------------------------------
+
+shareRideInfo ::
+  ShortId DM.Merchant ->
+  Id Common.Ride ->
+  Flow Common.ShareRideInfoRes
+shareRideInfo merchantShortId reqRideId = do
+  merchant <- findMerchantByShortId merchantShortId
+  let rideId = cast @Common.Ride @DRide.Ride reqRideId
+  ride <- runInReplica $ QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)
```

Comment:
we are doing these ride fetch and booking fetch query in `DRide.shareRideInfo` why are we doing again ?

Reply:
I have modified it

Reply:
Where have you modified it?

Reply:
I don't see any repetition in `DRide.shareRideInfo`